File: HEUR-Trojan-Ransom.Win32.Generic-3c196e42e8c8c66394ac3f4569a3b1f1b1e956cf010e849feed22c50eb751796/clen.My/MyApplication.cs
using System;
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;
using System.Runtime.Serialization;
using System.Security;
using System.Windows.Forms;
using Microsoft.VisualBasic.ApplicationServices;
using ns0;

namespace clen.My;

[GeneratedCode("MyTemplate", "11.0.0.0")]
[EditorBrowsable(EditorBrowsableState.Never)]
internal sealed class MyApplication : WindowsFormsApplicationBase
{
	private static List<WeakReference> list_0;

	[SecuritySafeCritical]
	static MyApplication()
	{
		Class6.smethod_1();
		list_0 = new List<WeakReference>();
	}

	private static void smethod_0(object object_0)
	{
		List<WeakReference> gparam_ = list_0;
		bool bool_ = false;
		checked
		{
			try
			{
				Class3.Class5.smethod_0((object)gparam_, ref bool_, (short)838, 855);
				if (list_0.Count == list_0.Capacity)
				{
					int num = 0;
					int num2 = list_0.Count - 1;
					int num3 = 0;
					while (true)
					{
						int num4 = num3;
						int num5 = num2;
						if (num4 > num5)
						{
							break;
						}
						WeakReference gparam_2 = (WeakReference)(ISerializable)list_0[num3];
						if (Class3.Class5.smethod_2(gparam_2, 'ʇ', 713))
						{
							if (num3 != num)
							{
								list_0[num] = list_0[num3];
							}
							num++;
						}
						num3++;
					}
					list_0.RemoveRange(num, list_0.Count - num);
					list_0.Capacity = list_0.Count;
				}
				list_0.Add(new WeakReference(Class6.smethod_2(object_0, '\u0015', 52)));
			}
			finally
			{
				if (bool_)
				{
					Class10.Class13.smethod_5((object)gparam_, 'Á', '\u0082');
				}
			}
		}
	}

	[MethodImpl(MethodImplOptions.NoOptimization)]
	[EditorBrowsable(EditorBrowsableState.Advanced)]
	[STAThread]
	[SecuritySafeCritical]
	internal static void Main(string[] args)
	{
		Class6.smethod_1();
		try
		{
			Class3.smethod_7(WindowsFormsApplicationBase.get_UseCompatibleTextRendering(), 42, 'd');
		}
		finally
		{
		}
		Class3.smethod_8<WindowsFormsApplicationBase>((WindowsFormsApplicationBase)(object)MyProject.MyApplication_0, args, (short)781, (short)839);
	}

	public MyApplication()
		: base((AuthenticationMode)0)
	{
		smethod_0(this);
		((WindowsFormsApplicationBase)this).set_IsSingleInstance(false);
		((WindowsFormsApplicationBase)this).set_EnableVisualStyles(true);
		Class3.Class5.smethod_3<WindowsFormsApplicationBase>((WindowsFormsApplicationBase)(object)this, bool_0: true, '\u02f0', (short)644);
		((WindowsFormsApplicationBase)this).set_ShutdownStyle((ShutdownMode)0);
	}

	protected override void OnCreateMainForm()
	{
		((WindowsFormsApplicationBase)this).set_MainForm((Form)(object)MyProject.Class0_0.method_0());
	}
}

File: Trojan-Ransom/Win32/E/Trojan-Ransom.Win32.Encoder.sgl-7a880aba1da9729c1282dde0ab44ec64fe371f453446d4bb613bf28db4d9cf6f/Jigsaw/My/MyApplication.vb
Imports System
Imports System.CodeDom.Compiler
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms
Imports Microsoft.VisualBasic.ApplicationServices

Namespace Jigsaw.My
	' Token: 0x02000002 RID: 2
	<GeneratedCode("MyTemplate", "11.0.0.0")>
	<EditorBrowsable(EditorBrowsableState.Never)>
	Friend Class MyApplication
		Inherits WindowsFormsApplicationBase

		' Token: 0x06000001 RID: 1 RVA: 0x00002638 File Offset: 0x00000838
		<STAThread()>
		<EditorBrowsable(EditorBrowsableState.Advanced)>
		<DebuggerHidden()>
		<MethodImpl(MethodImplOptions.NoOptimization)>
		Friend Shared Sub Main(args As String())
			Try
				Application.SetCompatibleTextRenderingDefault(WindowsFormsApplicationBase.UseCompatibleTextRendering)
			Finally
			End Try
			MyProject.Application.Run(args)
		End Sub

		' Token: 0x06000002 RID: 2 RVA: 0x00002050 File Offset: 0x00000250
		<DebuggerStepThrough()>
		Public Sub New()
			MyBase.New(AuthenticationMode.Windows)
			MyBase.IsSingleInstance = False
			MyBase.EnableVisualStyles = True
			MyBase.SaveMySettingsOnExit = True
			MyBase.ShutdownStyle = ShutdownMode.AfterMainFormCloses
		End Sub

		' Token: 0x06000003 RID: 3 RVA: 0x00002075 File Offset: 0x00000275
		<DebuggerStepThrough()>
		Protected Overrides Sub OnCreateMainForm()
			MyBase.MainForm = MyProject.Forms.Form2
		End Sub
	End Class
End Namespace
